presto-orc/src/main/java/com/facebook/presto/orc/OrcWriteValidation.java

```diff
@@ -236,35 +255,146 @@ public void validateRowGroupStatistics(OrcDataSourceId orcDataSourceId, long str
             RowGroupStatistics expectedRowGroup = expectedRowGroupStatistics.get(rowGroupIndex);
             if (expectedRowGroup.getValidationMode() != HASHED) {
                 Map<Integer, ColumnStatistics> expectedStatistics = expectedRowGroup.getColumnStatistics();
-                Set<Integer> actualColumns = actualRowGroupStatistics.keySet().stream()
-                        .map(StreamId::getColumn)
-                        .collect(Collectors.toSet());
-                if (!expectedStatistics.keySet().equals(actualColumns)) {
+                Map<Integer, ColumnStatistics> actualStatistics = aggregateRowGroupStatisticsFromRowIndex(orcDataSourceId, actualRowGroupStatistics, stripeOffset, rowGroupIndex);
+
+                // remove empty row group stats for empty flat maps
+                expectedStatistics = adjustRowGroupStatisticsForFlatMaps(orcDataSourceId, stripeOffset, expectedStatistics);
```

Comment:
Remove zero-values stats for value nodes for maps containing all null/empty elements. Do this after a bit of cross validation with the stripe column stats.